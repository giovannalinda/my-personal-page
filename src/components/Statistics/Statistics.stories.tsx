import { Meta, Story } from '@storybook/react'
import { Statistics } from './Statistics'

export default {
  title: 'Components/Statistics',
  component: Statistics,
} as Meta

export const Default: Story = () => <Statistics />
