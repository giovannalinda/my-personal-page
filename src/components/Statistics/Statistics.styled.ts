import styled from 'styled-components'
import { theme } from '~/styles'

export const List = styled.ul`
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${theme.colors.gray900};
`

export const ListItem = styled.li`
  padding-bottom: ${theme.space[7]};
  margin-bottom: ${theme.space[2]};

  & + li {
    padding-left: 8rem;
  }
`

export const Title = styled.h1`
  padding-top: ${theme.space[7]};
  font-size: ${theme.font.sizes.large};
  font-weight: ${theme.font.weigths.regular};
  padding-bottom: ${theme.space[4]};
`

export const Paragraph = styled.p`
  font-size: ${theme.font.sizes.verySmall};
`
