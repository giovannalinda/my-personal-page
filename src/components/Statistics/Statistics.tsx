import { useTranslation } from '~/lib'

import * as S from './Statistics.styled'

export function Statistics() {
  const { t } = useTranslation()

  return (
    <S.List>
      <S.ListItem>
        <S.Title>2+</S.Title>
        <S.Paragraph>{t('Years Experience')}</S.Paragraph>
      </S.ListItem>

      <S.ListItem>
        <S.Title>600+</S.Title>
        <S.Paragraph>{t('Contributions This Year')}</S.Paragraph>
      </S.ListItem>

      <S.ListItem>
        <S.Title>100%</S.Title>
        <S.Paragraph>{t('Of Satisfaction')}</S.Paragraph>
      </S.ListItem>
    </S.List>
  )
}
