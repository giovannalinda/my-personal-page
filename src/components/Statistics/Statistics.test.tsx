import { render } from '~/__helpers__/app-tests'
import { Statistics } from './Statistics'

describe('Statistics', () => {
  it('matches snapshot', () => {
    const { container } = render(<Statistics />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
