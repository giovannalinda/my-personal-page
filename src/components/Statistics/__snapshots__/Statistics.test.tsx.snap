// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics matches snapshot 1`] = `
<ul
  class="sc-bczRLJ eQsOC"
>
  <li
    class="sc-gsnTZi dkCecp"
  >
    <h1
      class="sc-dkzDqf hKJCPV"
    >
      2+
    </h1>
    <p
      class="sc-hKMtZM SUDTQ"
    >
      Years Experience
    </p>
  </li>
  <li
    class="sc-gsnTZi dkCecp"
  >
    <h1
      class="sc-dkzDqf hKJCPV"
    >
      600+
    </h1>
    <p
      class="sc-hKMtZM SUDTQ"
    >
      Contributions This Year
    </p>
  </li>
  <li
    class="sc-gsnTZi dkCecp"
  >
    <h1
      class="sc-dkzDqf hKJCPV"
    >
      100%
    </h1>
    <p
      class="sc-hKMtZM SUDTQ"
    >
      Of Satisfaction
    </p>
  </li>
</ul>
`;
